modules/shapefile/src/lib/parsers/parse-shp.js

```diff
@@ -85,10 +98,13 @@ function parseState(state, result = {}, binaryReader) {
 
         case STATE.EXPECTING_HEADER:
           // Parse initial file header
-          const dataView = binaryReader.getDataView(SHP_HEADER_SIZE, 'SHP header');
-          if (!dataView) {
+          // Make sure bytes exist
+          if (!binaryReader.hasAvailableBytes(SHP_HEADER_SIZE)) {
```

Comment:
This wasn't previously an issue, because the first batch had at least 100 bytes, but still a good check.

Reply:
Well the previous case returning `null` should have had the same semantics, at least that was the intention...

Reply:
Yes but with the current setup, `binaryReader.getDataView` never returns `null`, it either returns data or throws an error. So `hasAvailableBytes` should always be called first.

Reply:
OK. The idea was that it would return null if there was not enough data, except if `end()` had been called, in which case it would throw an error.

I am fine with whatever API makes for the cleanest parser code.

Reply:
I changed this in `getDataView` to return `null` when not enough data except when `this.ended` is true.